entrypoint

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+set -eu -o pipefail
+
+say() {
+	echo -e "${@}"
+}
+
+fail() {
+	say "${@}" 1>&2
+	exit ${EXIT_CODE:-1}
+}
+
+urlencode() {
+	local STR="${1}"
+	local LEN=${#STR}
+	local RESULT=""
+	local POS C O
+
+	for (( POS=0 ; POS < LEN ; POS++ )); do
+		C=${STR:$POS:1}
+		case "${C}" in
+			[-_.~a-zA-Z0-9/] )	O="${C}" ;;
+			* )					printf -v O '%%%02x' "'${C}"
+		esac
+		RESULT+="${O}"
+	done
+	echo "${RESULT}"    # You can either set a return variable (FASTER) 
+}
+
+#
+# TODO: Do we actually want to verify this?
```

Comment:
I think it makes sense to check these things.  Containers should be immutable, but they are not really formally immutable.